agent_guard_core/credentials/gcp_secrets_manager_provider.py

```diff
@@ -0,0 +1,145 @@
+
+from .secrets_provider import BaseSecretsProvider, SecretProviderException
+
+DEFAULT_PROJECT_ID = "default"
+DEFAULT_SECRET_ID = "agentic_env_vars"
+DEFAULT_SECRET_VERSION = "latest"
+DEFAULT_REPLICATION_TYPE = "automatic"
+SUPPORTED_REPLICATION_TYPES = ["automatic", "user_managed"]
+
+
+class GCPSecretsProvider(BaseSecretsProvider):
+
+    def __init__(self,
+                 project_id: str = DEFAULT_PROJECT_ID,
+                 secret_id: str = DEFAULT_SECRET_ID,
+                 secret_id_version: str = DEFAULT_SECRET_VERSION,
+                 region: Optional[str] = None,
+                 replication_type: str = DEFAULT_REPLICATION_TYPE):
+        super().__init__()
+        self._project_id = project_id
+        self._secret_id = secret_id
+        self._secret_id_version = secret_id_version
+        self._region = region
+        self._client = None
+
+        if replication_type not in SUPPORTED_REPLICATION_TYPES:
+            raise SecretProviderException(
+                f"Unsupported replication type: {replication_type}. "
+                f"Supported types are: {', '.join(SUPPORTED_REPLICATION_TYPES)}"
+            )
+        self._replication_type = replication_type
+
+    def connect(self) -> bool:
+        if self._client:
+            return True
+        try:
+            self._client = secretmanager.SecretManagerServiceClient()
+            return True
+        except Exception as e:
+            self.logger.error("Error initializing Secret Manager client: %s",
+                              e)
+            raise SecretProviderException(
+                f"GCP Secret Manager init failed: {e}") from e
+
+    def _get_secret_path(self) -> str:
+        if self._region is not None:
+            return f"projects/{self._project_id}/locations/{self._region}/secrets/{self._secret_id}"
+        return f"projects/{self._project_id}/secrets/{self._secret_id}"
+
+    def _get_version_path(self) -> str:
+        return f"{self._get_secret_path()}/versions/{self._secret_id_version}"
+
+    def _get_secret_parent(self) -> str:
+        return f"projects/{self._project_id}"
+
+    def get_secret_dictionary(self) -> Dict[str, str]:
+        self.connect()
+        try:
+            version_path = self._get_version_path()
+            response = self._client.access_secret_version(
+                request={"name": version_path})
+            secret_text = response.payload.data.decode("utf-8")
+            return json.loads(secret_text)
+        except NotFound:
+            self.logger.warning("Secret not found: %s", self._secret_id)
+            return {}
+        except Exception as e:
+            self.logger.error("Failed to retrieve secret:%s", e)
+            raise SecretProviderException(
+                f"Error retrieving secret: {e}") from e
+
+    def store_secret_dictionary(self, secret_dictionary: Dict[str,
+                                                              str]) -> None:
+        if not secret_dictionary:
```

Comment:
In the [AWS provider](https://github.com/oryandrive/agent-guard/blob/feature/support-gcp-secret-manager/agent_guard_core/credentials/aws_secrets_manager_provider.py#L93), the check is specifically for `None`, which means it will accept an empty dictionary of `{}`. This code doesn't accept an empty dictionary which makes [this test](https://github.com/oryandrive/agent-guard/blob/feature/support-gcp-secret-manager/tests/integration/gcp_secret_manager_provider_test.py#L23) fail in a clean environment, where the tested GCP secret is empty. Can you please fix the issue?
```suggestion
        # an empty dictionary is valid, yet a none one is not and raises an exception
        if secret_dictionary is None:
```

Reply:
Thanks, it's fixed now 😊
Let me know if anything else is needed.
I've added documentation and two test cases to ensure both None and empty dictionaries behave as expected.